glomap/estimators/relpose_estimation.cc

```diff
@@ -1,57 +1,64 @@
 #include "glomap/estimators/relpose_estimation.h"
 
+#include <iomanip>
+
 #include <PoseLib/robust.h>
 
 namespace glomap {
+
 void EstimateRelativePoses(ViewGraph& view_graph,
                            std::unordered_map<camera_t, Camera>& cameras,
                            std::unordered_map<image_t, Image>& images,
                            const RelativePoseEstimationOptions& options) {
-  std::vector<image_pair_t> image_pair_ids;
+  std::vector<image_pair_t> valid_pair_ids;
   for (auto& [image_pair_id, image_pair] : view_graph.image_pairs) {
     if (!image_pair.is_valid) continue;
-    image_pair_ids.push_back(image_pair_id);
+    valid_pair_ids.push_back(image_pair_id);
   }
 
-  image_pair_t inverval = std::ceil(image_pair_ids.size() / 10.);
-  LOG(INFO) << "Estimating relative pose for " << image_pair_ids.size()
-            << " pairs";
-  for (image_pair_t chunks = 0; chunks < 10; chunks++) {
-    std::cout << "\r Estimating relative pose: " << chunks * 10 << "%"
-              << std::flush;
-    image_pair_t start = chunks * inverval;
-    image_pair_t end = std::min((chunks + 1) * inverval, image_pair_ids.size());
-#pragma omp parallel for
-    for (image_pair_t pair = start; pair < end; pair++) {
-      ImagePair& image_pair = view_graph.image_pairs[image_pair_ids[pair]];
-      image_t idx1 = image_pair.image_id1;
-      image_t idx2 = image_pair.image_id2;
+  const size_t num_valid_pairs = valid_pair_ids.size();
+  LOG(INFO) << "Estimating relative pose for " << num_valid_pairs << " pairs";
+
+#pragma omp parallel
+  {
+    // Define variables outside parallel for loop to reuse allocations.
+    std::vector<Eigen::Vector2d> points2D_1, points2D_2;
+    std::vector<char> inliers;
```

Comment:
@lpanaf Is it intentional that we don't update the inliers of the image_pair here but keep the original colmap one?